presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java

```diff
@@ -85,19 +87,22 @@ public Iterator<HiveFileInfo> list(
             HiveDirectoryContext hiveDirectoryContext)
     {
         RuntimeStats runtimeStats = hiveDirectoryContext.getRuntimeStats();
+        long startTime = System.nanoTime();
         if (hiveDirectoryContext.isCacheable()) {
             // DO NOT USE Caching, when cache is disabled.
             // This is useful for debugging issues, when cache is explicitly disabled via session property.
             List<HiveFileInfo> files = cache.getIfPresent(path);
             if (files != null) {
                 runtimeStats.addMetricValue(DIRECTORY_LISTING_CACHE_HIT, NONE, 1);
+                runtimeStats.addMetricValue(DIRECTORY_LISTING_TIME_NANOS, NANO, System.nanoTime() - startTime);
```

Comment:
```
import static io.airlift.units.Duration.nanosSince;

runtimeStats.addMetricValue(DIRECTORY_LISTING_TIME_NANOS, NANO,nanosSince(startTime));
```

Reply:
nanosSince returns Duration, so we need one extra step to convert it to long

Reply:
hmm. should be fine to leave it like this then.